### PR TITLE
Tkt 00184 - Update header layout

### PR DESCRIPTION
**Pull request by Android63**
Updated header from this:
screen shot 2018-01-29 at 13 12 34

To this:
screen shot 2018-01-29 at 13 14 31

The main differences are:

1. The search bar is optional but, if present, is 100% of its container width and doesn't push the menu bar down.
2. The menu bar is left-aligned in its container
3. The login button, if present, is right-aligned in its container.

Tested in various browsers, including mobile. In some older browsers, the mobile view is used (which is a reasonable failsafe).

### CODE REVIEW

File: components/nav-primary/_components.nav-primary.scss [hunk 1]
```diff
@@ -14,8 +14,9 @@
 
   @include mq($from: md) {
     font-size: 0; /* [1] */
-    float: right;
+    float: left;
     clear: none;
+    margin-top: $global-border-mid+36;
```
Comment (cehwitham) [inline components/nav-primary/_components.nav-primary.scss]: Where does the 36 come from here @Android63?

File: components/search-bar/_components.search-bar.scss [hunk 2]
```diff
@@ -9,13 +9,22 @@
    }
    .c-search__input {
      @extend .c-form-input;
-     height: 38px;
+     height: 36px;
```
Comment (cehwitham) [inline components/search-bar/_components.search-bar.scss]: Has this always been a fixed height?

File: components/search-bar/_components.search-bar.scss [hunk 3]
```diff
@@ -9,13 +9,22 @@
...
      border-radius: 0;
      background-color: $color-nav-primary-trigger-bg;
+   }
+   
+   /* Apply negative bottom margin to search bar in header
+   so that, when it is present, it doesn't push the menu bar down
+   */
+   .c-page-header__controls > .c-search {
+     @include mq($from: md) {
+       margin-bottom: -36px;
```
Comment (cehwitham) [inline components/search-bar/_components.search-bar.scss]: @csswizardry any thoughts on this? Is this the best way to accomplish this?

File: index.html [hunk 4]
```diff
@@ -104,7 +104,7 @@
     <div class="o-wrapper  c-page-header__inner">
 
       <a href="#0" class="c-page-header__lockup">
-        <img src="./assets/img/logos/nhs-leadership-academy.png" alt="NHS Leadership Academy" width="204" height="62" class="c-page-header__logo" />
+        <img src="./assets/img/logos/NHS_LA_Logo_Blue.png" alt="NHS Leadership Academy" class="c-page-header__logo" />
```
Comment (cehwitham) [inline index.html]: @Android63 should we have a more standardised filenaming convention for images? Seems we're switching between - and _ and caps?

**Review by cehwitham — CHANGES_REQUESTED** [review]

File: components/nav-primary/_components.nav-primary.scss (hunk 1 above)
Comment (Android63) [reply components/nav-primary/_components.nav-primary.scss]: It's the height of the search bar (which has a bottom margin of -36px to cancel this out). I've added a comment to that effect.

File: components/search-bar/_components.search-bar.scss (hunk 2 above)
Comment (Android63) [reply components/search-bar/_components.search-bar.scss]: I think so. The height has to match the NHS lozenge in the logo. I've added a comment to that effect.

File: index.html (hunk 4 above)
Comment (Android63) [reply index.html]: Agreed. This is addressed in https://github.com/NHSLeadership/nightingale/pull/179

**Comment by Android63** [comment]
Are you OK with me merging this in @cehwitham? I will ask Harry to review, but I'd like to get the functionality in place for now and correct any issues he raises later.

File: components/search-bar/_components.search-bar.scss (hunk 3 above)
Comment (csswizardry) [reply components/search-bar/_components.search-bar.scss]: This is gonna be an interesting one to solve! Let’s discuss :)

Comment (Android63) [reply components/search-bar/_components.search-bar.scss]: Looking forward to it :)

**Comment by Android63** [comment]
@cehwitham I've checked this locally and it looks fine. Could you review it with a view to a merge please?

**Review by cehwitham — APPROVED** [review]

**Pull request merged**